File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/AgentFactory.java
package org.akoshterek.backgammon.agent;

import org.akoshterek.backgammon.agent.gnubg.GnubgAgent;
import org.akoshterek.backgammon.agent.pubeval.PubEvalAgent;
import org.akoshterek.backgammon.agent.raw.RawBatch40;
import org.akoshterek.backgammon.agent.raw.RawRl40;
import org.akoshterek.backgammon.eval.Evaluator;

/**
 * @author Alex
 *         date 02.08.2015.
 */
public class AgentFactory {
    public static Agent createAgent(String fullName) {
        String fullNameLower = fullName.toLowerCase();
        //String[] tokens = fullNameLower.split("-");

        Agent agent;
        switch (fullNameLower) {
            case "random":
                agent = new RandomAgent(Evaluator.getInstance().getBasePath());
                break;
            case "heuristic":
                agent = new HeuristicAgent(Evaluator.getInstance().getBasePath());
                break;
            case "pubeval":
                agent = new PubEvalAgent(Evaluator.getInstance().getBasePath());
                break;
            case "gnubg":
                agent = new GnubgAgent(Evaluator.getInstance().getBasePath());
                break;
            case "rawbatch40":
                agent = new RawBatch40(Evaluator.getInstance().getBasePath());
                break;
            case "rawrl40":
                agent = new RawRl40(Evaluator.getInstance().getBasePath());
                agent.load();
                break;
            default:
                throw new IllegalArgumentException("Unknown agent name " + fullName);
        }

        return agent;
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/AgentFactory.scala
package org.akoshterek.backgammon.agent

import org.akoshterek.backgammon.agent.gnubg.GnubgAgent
import org.akoshterek.backgammon.agent.pubeval.PubEvalAgent
import org.akoshterek.backgammon.agent.raw.RawTd40
import org.akoshterek.backgammon.eval.Evaluator
import org.akoshterek.backgammon.genetic.GeneticAgent
import org.akoshterek.backgammon.util.OptionsBean

import java.nio.file.Paths

object AgentFactory {
  def createAgent(fullName: String, options: OptionsBean): Agent = {
    val fullNameLower: String = fullName.toLowerCase
    val tokens = fullNameLower.split("-")

    tokens(0) match {
      case "random" =>
        new RandomAgent(Evaluator.basePath)
      case "heuristic" =>
        new HeuristicAgent(Evaluator.basePath)
      case "simpleheuristic" =>
        new SimpleHeuristicAgent(Evaluator.basePath)
      case "pubeval" =>
        PubEvalAgent(Evaluator.basePath)
      case "gnubg" =>
        new GnubgAgent(Evaluator.basePath)
      case "rawtd40" =>
        new RawTd40(Evaluator.basePath, options.alpha, options.lambda, options.gamma, options.experimentRunTag, isCopy = false, originalSeed = 16000000L,
                   useOutputBias = options.useOutputBias)

      case "geneticagent" =>
        GeneticAgent.fromPath(Evaluator.basePath, 6);


      case "geneticagent5" =>
        val path = Paths.get("experiments_ga/005_pubeval_5out")
        GeneticAgent.fromPath(path, 5);
      case "geneticagent6" =>
        val path = Paths.get("experiments_ga/006_pubeval_5out")
        GeneticAgent.fromPath(path, 6);

      case _ =>
        throw new IllegalArgumentException("Unknown agent name " + fullName)
    }
  }
}